src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs

```diff
@@ -200,7 +200,17 @@ public bool IsEmpty(DocumentId documentId)
                 foreach (var documentId in result.DocumentIds)
                 {
                     var document = project.GetDocument(documentId);
-                    Contract.ThrowIfNull(document);
+                    if (document == null)
+                    {
+                        // it can happen with build synchronization since, in build case, 
+                        // we don't have actual snapshot (we have no idea what sources out of proc build has picked up)
+                        // so we might be out of sync.
+                        // example of such cases will be changing anything about solution while building is going on.
+                        // it can be user explict actions such as unloading project, deleting a file, but also it can be 
+                        // something project system or roslyn workspace does such as populating workspace right after
+                        // solution is loaded.
+                        continue;
```

Comment:
If any of the above is true, should we be saving the data?  It sounds like it might be  better to not save as we might be saving invalid data wrt the state of the project.